src/core/cm/unitconfig/unitconfig.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (C) 2025 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <core/common/tools/fs.hpp>
+#include <core/common/tools/logger.hpp>
+#include <core/common/tools/semver.hpp>
+
+#include "unitconfig.hpp"
+
+namespace aos::cm::unitconfig {
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+Error UnitConfig::Init(const String& cfgFile, nodeinfoprovider::NodeInfoProviderItf& nodeInfoProvider,
+    NodeConfigHandlerItf& nodeConfigHandler, JSONProviderItf& jsonProvider)
+{
+    LOG_INF() << "Init unit config";
+
+    mCfgFile           = cfgFile;
+    mNodeInfoProvider  = &nodeInfoProvider;
+    mNodeConfigHandler = &nodeConfigHandler;
+    mJSONProvider      = &jsonProvider;
+
+    return LoadConfig();
```

Comment:
We should not return error if load fails, otherwise whole CM will be restarted.  Just set err in unit config status and display error log.